src/python/pants/option/options_test.py

```diff
@@ -443,28 +448,6 @@ def register(opts: Options) -> None:
     assert vals2.qux == "uu"
 
 
-def test_scope_deprecation_default_config_section(caplog) -> None:
```

Comment:
This was testing emergent behavior, not API-defined behavior. Implementing this behavior in the native parser would be tricky and IMO not worth it:

A) It's not at all clear that this is the correct behavior. Generally we treat a config entry in DEFAULT as-if it was in the relevant section, so special-casing this seems unnecessary.
B) This behavior is only triggered when we deprecate an entire scope, which we haven't done in a while, and currently have no such deprecations pending, so no user will see any effect.